### PR TITLE
GTT-1209: Update dashboard editor backgrounds' color

### PR DESCRIPTION
**Pull request by ghost**
## Description

In the pages where users edit, create or publish dashboards or dashboard content we wanted to enhance the appearance of the canvas' (rectangular boxes that are white in color and have shadow effects on the edges), so I changed the page background colors of the the Add Content/Image/Text/Metric/Table.tsx, Edit Chart/Dashboard/Details/Image/Metrics/Table/Text.tsx, CreateDashboard.tsx, and PublishDashboard.tsx files/pages by importing a custom hook I made in frontend/src/hooks/background-hooks. In order to do this:

1.  In the custom hooks file, I used the UseEffects Hook to change the color of a page #fafafa and restore it back to its original color when a user navigates away.
2. In the files mentioned above, I called the library I created

Check out https://aakbalaj.badger.wwps.aws.dev/

## Testing

I tested my changes by running the badger website on my machine as well as via AWS, navigating to the pages mentioned above, and checking if the pages background colors changed to #fafafa (light greyish color). One can also test this by doing console.log(document.body.style.backgroundColor)and verify whether the background color changed to rgb(250, 250, 250). I finally ran npm run test and ./test.sh to verify if all test cases are working. 

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ajmokotoff — CHANGES_REQUESTED** [review]
This is a great start. I don't think we should be changing the package.json or the package-lock files for this change. Can we remove the following changes?

- cdk/package-lock
- cdk/package.json
- /package-lock
- selected

As for the actual change, I think this is a great design. I wonder though, do we think we could maybe put this logic into a lib, so we could call the same code instead of having the same code everywhere?

File: cdk/package-lock.json [hunk 1]
```diff
@@ -4713,4 +4713,4 @@
       "dev": true
     }
   }
-}
+}
```
Comment (ajmokotoff) [inline cdk/package-lock.json]: Can we remove this change?

File: cdk/package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "build": "tsc",
     "watch": "tsc -w",
     "test": "jest --passWithNoTests",
-    "cdk": "cdk",
+    "cdk": "cdk",
```
Comment (ajmokotoff) [inline cdk/package.json]: Can we get rid of this as well?

File: frontend/src/containers/AddContent.tsx [hunk 3]
```diff
@@ -1,11 +1,12 @@
-import React from "react";
+import React, { useEffect } from "react";
```
Comment (ajmokotoff) [inline frontend/src/containers/AddContent.tsx]: Unused import, now that you refactored

File: frontend/src/containers/AddImage.tsx [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
```
Comment (ajmokotoff) [inline frontend/src/containers/AddImage.tsx]: Unused import

File: frontend/src/containers/AddMetric.tsx [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
```
Comment (ajmokotoff) [inline frontend/src/containers/AddMetric.tsx]: Unused import

File: frontend/src/containers/AddMetrics.tsx [hunk 6]
```diff
@@ -34,6 +34,7 @@ import Alert from "../components/Alert";
 import PrimaryActionBar from "../components/PrimaryActionBar";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faSearch } from "@fortawesome/free-solid-svg-icons";
+import { ChangeBackgroundColor } from "../hooks/background-hooks";
```
Comment (ajmokotoff) [inline frontend/src/containers/AddMetrics.tsx]: Unused import

File: frontend/src/containers/AddText.tsx [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
```
Comment (ajmokotoff) [inline frontend/src/containers/AddText.tsx]: Unused import

File: frontend/src/containers/CreateDashboard.tsx [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import React from "react";
+import React, { useEffect } from "react";
```
Comment (ajmokotoff) [inline frontend/src/containers/CreateDashboard.tsx]: Unused import

File: frontend/src/containers/EditDashboard.tsx [hunk 9]
```diff
@@ -1,4 +1,4 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
```
Comment (ajmokotoff) [inline frontend/src/containers/EditDashboard.tsx]: Unused import

File: frontend/src/containers/EditDetails.tsx [hunk 10]
```diff
@@ -1,4 +1,4 @@
-import React from "react";
+import React, { useEffect } from "react";
```
Comment (ajmokotoff) [inline frontend/src/containers/EditDetails.tsx]: Unused import

File: frontend/src/containers/EditImage.tsx [hunk 11]
```diff
@@ -1,4 +1,4 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
```
Comment (ajmokotoff) [inline frontend/src/containers/EditImage.tsx]: Unused import

File: frontend/src/containers/EditText.tsx [hunk 12]
```diff
@@ -1,4 +1,4 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
```
Comment (ajmokotoff) [inline frontend/src/containers/EditText.tsx]: Unused import

File: frontend/src/containers/PublishDashboard.tsx [hunk 13]
```diff
@@ -1,4 +1,4 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
```
Comment (ajmokotoff) [inline frontend/src/containers/PublishDashboard.tsx]: Unused import

**Review by ajmokotoff — CHANGES_REQUESTED** [review]
You are close. Some small nitpicks:

- There were a few minor changes (line breaks, extra spaces) added to the cdk/package-lock.json and cdk/package.jsonfiles. Lets remove those
- Now that you have refactored your logic into a separate module, you should remove the import useEffect that you added, since it is no longer needed

**Review by ajmokotoff — APPROVED** [review]
This looks good! Great job Aakash.

File: frontend/src/hooks/background-hooks.ts [hunk 14]
```diff
@@ -0,0 +1,17 @@
+import React, { useEffect } from "react";
+
+export function ChangeBackgroundColor(): any {
```
Comment (miguelpdiaz8) [inline frontend/src/hooks/background-hooks.ts]: By convention, our hooks start with "use" in lowercase like "useChangeBackgroundColor"

**Pull request merged**